db_master_deploy/deploy.sh

```diff
@@ -0,0 +1,331 @@
+    fi
+
+    echo "truncate table $target_id"
+    ( psql -U pgkogis -h localhost -c "begin; TRUNCATE TABLE $target_schema.$target_table; commit;" -d "$target_db" )
+
+    (
+    for i in $(eval echo {1..$jobs})
+    do 
+        offset=$(echo "(($i-1)*$increment)" | bc)
+        if [ $((offset+$increment)) -gt $rows ]; then counter=$rows; else counter=$((offset+$increment));fi
+        echo "dumping $offset..$counter"
+        ( psql -U pgkogis -h localhost -qAt -d $source_db -c "COPY ( SELECT * FROM $source_schema.$source_table order by ${primary_keys:=1} asc offset $offset limit $increment ) TO STDOUT with csv" | psql -U pgkogis -h localhost -qAt -d $target_db -c "SET session_replication_role = replica; COPY $target_schema.$target_table from stdin with csv; SET session_replication_role = DEFAULT;" )& pids="$pids $!"
+    done;  
+    wait $pids 2> /dev/null
+    )
+
+    echo "create indexes on $target_id"
+    ( pg_dump -U pgkogis -h localhost --if-exists -c -t $source_schema.$source_table -s $source_db 2>/dev/null | egrep -i "\bcreate\b" | egrep -i "\bindex\b" | sed "s/^/set search_path = $source_schema, public, pg_catalog; /" | sed "s/'/\\\'/g" | xargs --max-procs=$jobs -I '{}' sh -c 'psql -U pgkogis -h localhost -d $@ -c "{}"' -- "$target_db" ) 
+
+    if [ "${#foreign_keys[@]}" -gt 0 ]
+    then
+        for i in "${!foreign_keys[@]}"
+        do
+            echo "CREATE FOREIGN KEY CONSTRAINT $i ON $target_id ..."
+            echo "ALTER TABLE ONLY $target_schema.$target_table ADD CONSTRAINT $i ${foreign_keys[$i]};" | psql -U pgkogis -h localhost -d $target_db &> /dev/null
+        done
+    fi
+}
+
+echo "start ${0##*/} $* (pid: $$)"
+CPUS=`(grep "processor" < /proc/cpuinfo | wc -l) 2>/dev/null` || CPUS=1
+START=$(date +%s%3N)
+
+# check for mandatory arguments 
+if [[ -z "$source_objects" || -z "$target" ]];
+then
+    echo "missing a required parameter (source_db -s and staging -t are required)" >&2
+    exit 1
+fi
+
+# check if we have a valid target
+if [[ ! $targets == *$target* ]]
+then
+    echo "valid deploy targets are: '$targets'" >&2
+    exit 1
+fi
+
+# check db access
+if [[ -z `psql -lqt -h localhost -U pgkogis` ]]; then
+    echo "Unable to connect to database cluster" >&2
+    exit 1
+fi
+
+# check for lockfile
+(set -o noclobber; echo "$locktext" > "$lockfile"  2> /dev/null) || { echo "lockfile found: $lockfile '$(cat $lockfile)'" >&2; exit 1; }
+
+trap 'rm -f "$lockfile"; echo "trap: script aborted" >&2; exit $?' INT TERM EXIT
+
+attached_slaves=$(psql -qAt -h localhost -d postgres -U pgkogis -c "select count(1) FROM pg_stat_replication where state='streaming';")
+
+# loop through source_object values
+for source_object in "${array_source[@]}"; do
+    array=(${source_object//./ })
+    # check source objects
+    if [ "${#array[@]}" -ne "3" -a "${#array[@]}" -ne "1" ]; then
+        echo "table data sources have to be formatted like this: db.schema.table, database sources like this: db" >&2
+        exit 1
```

Comment:
Yes, but it is already the case now, and with the exception of the `bod`, the relation bewtween tables are pretty loose.